### PR TITLE
Detach custom event listener

### LINKED ISSUE #370: Tooltip hides immediately
Hey,
I am confused because my tooltip hides immediately and I don't understand why. You can watch it [here](https://www.screencast.com/t/2Llt30hV)
My code:

```
.map((node, index) => {
	const arrow = node.visualBlock as TreeArrow;
	const id = `arrow-menu-${node.visualBlock.blockId}`;
	return (
		<ReactTooltip
			id={id}
			getContent={this.getMenuItems(arrow)}
			effect='solid'
			place='bottom'
			key={index}
			className='menu-buttons'
			type='light'
			event='click'
			globalEventOff='click'
			isCapture={true}
		/>
	);
});
```
```
export const PlusButton = (props: PlusButtonProps) => {
	const yPosition = props.layoutObject.position.size.height / 2 - height / 2;
	const dateFor = `arrow-menu-${props.layoutObject.visualBlock.blockId}`;
	return (
		<svg
			className='plus-button-block'
			width={width}
			height={height}
			viewBox={`0 0 ${width} ${height}`}
			x={xPosition}
			y={yPosition}
			data-for={dateFor}
			data-tip={true}
			data-offset={tooltipOffset}
		>
		   ...
		</svg>
	);
};
```
Why I am getting in this hide function ?

### PR DESCRIPTION
**Pull request by P0lip**
Fixes #370 

This PR introduces usage of WeakMap, therefore it may be a breaking change, since WeakMap is unsupported in non-ES6 environments and requires a polyfill.
All projects using babel-polyfill should be unaffected.

### CODE REVIEW

File: src/decorators/customEvent.js [hunk 1]
```diff
@@ -33,7 +33,27 @@ const setUntargetItems = function (currentTarget, targetArray) {
   }
 }
 
-let customListener
+const customListeners = {
+  registry: new WeakMap(),
```
[the remark this answers is not in the document]
Comment (P0lip) [reply src/decorators/customEvent.js]: We can define a property on given object. This is how WeakMap is usually polyfilled.

**Comment by aronhelser** [comment]
Great, the new approach looks reasonable. Can you please add a comment that this is essentially a polyfill for WeakMap? I'll see if I can try it out.

**Comment by P0lip** [comment]
Done.

**Pull request merged**